packages/syft/tests/syft/actions/action_object_test.py

```diff
@@ -0,0 +1,993 @@
+    assert action_result == expected
+
+    action_result = context.obj._syft_output_action_object(action_result)
+    assert isinstance(action_result, ActionObject)
+
+
+@pytest.mark.parametrize(
+    "testcase",
+    [
+        # (object, operation, *args, **kwargs)
+        ("abc", "capitalize", [], {}),
+        ("abc", "find", ["b"], {}),
+        (int(1), "__add__", [1], {}),
+        (float(1.2), "__add__", [1], {}),
+        (True, "__and__", [False], {}),
+        ((1, 1, 3), "count", [1], {}),
+        ([1, 2, 1], "count", [1], {}),
+        ([1, 2, 3], "append", [4], {}),
+        ({"a": 1, "b": 2}, "update", [{"c": 3}], {}),
+        (set({1, 2, 3, 3}), "add", [5], {}),
+        (set({1, 2, 3, 3}), "clear", [], {}),
+    ],
+)
+def test_actionobject_syft_make_action(worker, testcase):
+    orig_obj, op, args, kwargs = testcase
+
+    obj = helper_make_action_obj(orig_obj)
+    args, kwargs = helper_make_action_args(*args, **kwargs)
+    obj_pointer, args_pointers, kwargs_pointers = helper_make_action_pointers(
+        worker, obj, *args, **kwargs
+    )
+
+    path = str(type(orig_obj))
+    action = obj.syft_make_action(path, op, args=args_pointers, kwargs=kwargs_pointers)
+
+    assert action.full_path.endswith("." + op)
+
+
+@pytest.mark.parametrize(
+    "testcase",
+    [
+        # (object, operation, *args, **kwargs)
+        ("abc", "capitalize", [], {}),
+        ("abc", "find", ["b"], {}),
+        (int(1), "__add__", [1], {}),
+        (float(1.2), "__add__", [1], {}),
+        (True, "__and__", [False], {}),
+        ((1, 1, 3), "count", [1], {}),
+        ([1, 2, 1], "count", [1], {}),
+        ([1, 2, 3], "append", [4], {}),
+        ({"a": 1, "b": 2}, "update", [{"c": 3}], {}),
+        (set({1, 2, 3, 3}), "add", [5], {}),
+        (set({1, 2, 3, 3}), "clear", [], {}),
+    ],
+)
+def test_actionobject_syft_make_method_action(worker, testcase):
+    orig_obj, op, args, kwargs = testcase
+
+    obj = helper_make_action_obj(orig_obj)
+    args, kwargs = helper_make_action_args(*args, **kwargs)
+    obj_pointer, args_pointers, kwargs_pointers = helper_make_action_pointers(
+        worker, obj, *args, **kwargs
+    )
+
+    action = obj.syft_make_method_action(op, args=args_pointers, kwargs=kwargs_pointers)
+
+    assert action.full_path.endswith("." + op)
+
+
+@pytest.mark.parametrize(
+    "testcase",
+    [
+        # (object, operation, *args, **kwargs)
+        ("abc", "capitalize", [], {}),
+        ("abc", "find", ["b"], {}),
+        (int(1), "__add__", [1], {}),
+        (float(1.2), "__add__", [1], {}),
+        (True, "__and__", [False], {}),
+        ((1, 1, 3), "count", [1], {}),
+        ([1, 2, 1], "count", [1], {}),
+        ([1, 2, 3], "append", [4], {}),
+        ({"a": 1, "b": 2}, "update", [{"c": 3}], {}),
+        (set({1, 2, 3, 3}), "add", [5], {}),
+        (set({1, 2, 3, 3}), "clear", [], {}),
+    ],
+)
+def test_actionobject_syft_make_remote_method_action(worker, testcase):
```

Comment:
I dont think this method is used anywhere, maybe we should remove it. cc @madhavajay